src/components/TaskConfirmationModal/TaskConfirmationModal.jsx

```diff
@@ -216,25 +221,73 @@ export class TaskConfirmationModal extends Component {
-                         rows={2}
-                         cols="1"
-                         value={this.props.osmComment}
-                        onChange={e => this.props.setOSMComment(e.target.value)}
-                       />
+                     <div className="mr-border mr-border-gray-300 mr-shadow-md mr-p-4 mr-mt-4">
+                       <p className="mr-mb-2 mr-text-white mr-text-md mr-font-bold mr-bg-blue-600 mr-rounded">
+                         <FormattedMessage
+                           {...messages.osmUploadNotice }
+                         />
+                       </p>
+                       <div className="mr-text-base mr-mt-2 mr-text-yellow">
+                         <FormattedMessage {...messages.osmCommentHeader} />
+                       </div>
+
+                       <div>
+                         <textarea
+                           ref={this.commentInputRef}
+                           className="mr-input mr-text-white mr-placeholder-medium mr-bg-grey-lighter-10 mr-border-none mr-shadow-inner mr-p-3 mr-my-1"
+                           rows={2}
+                           value={this.props.osmComment}
+                           onChange={e => this.props.setOSMComment(e.target.value)}
+                         />
+                       </div>
+
+                       <div className="mr-bg-blue-dark shadow-md">
+                         <div className="mr-flex mr-justify-between">
+                           <ul className="mr-w-1/3 mr-px-2 mr-border mr-border-gray-300 mr-shadow-sm">
+                             <li className="mr-font-bold mr-py-1 mr-text-sm">Tag Name</li>
+                             {_get(this.props, 'tagDiffs[0]') && Object.keys(_get(this.props, 'tagDiffs[0]')).map(tagName => {
+                               const tagChange = _get(this.props, 'tagDiffs[0]')[tagName];
+                               if (['changed', 'removed', 'added'].includes(tagChange.status)) {
+                                 return (
+                                   <li key={tagName} className={classNames('mr-mb-2  mr-rounded', {
+                                     'mr-text-orange': tagChange.status === 'changed',
+                                     'mr-text-lavender-rose': tagChange.status === 'removed',
+                                     'mr-text-picton-blue': tagChange.status === 'added',
+                                   })}>
+                                     <strong className="mr-text-sm">{tagName}</strong>
+                                   </li>
+                                 );
+                               }
+                             })}
+                           </ul>
+                           <ul className="mr-w-1/3 mr-px-2 mr-border mr-border-gray-300 mr-shadow-sm"> 
+                             <li className="mr-font-bold mr-py-1 mr-text-sm">Current Value</li> 
```

Comment:
These column headers ("Tag Name", "Current Value", "Requested New Value") should probably all be localized.

Also, nitpick, but I think "Requested New Value" is confusing since the change is going to be published immediately (i.e. the user isn't submitting a "change request"). I'd go with "Tag name", "Old value", "New value" for clarity.